contracts/repository/blockchain/casino.go

```diff
@@ -170,6 +171,112 @@ func (r *CasinoBlockchainRepo) GetBonusBalances(casinos []*models.Casino, accoun
 	return bonusBalances, nil
 }
 
+func (r *CasinoBlockchainRepo) getTokenContractBalances(tokenContract eos.AccountName,
+	playerAccount eos.AccountName) ([]eos.Asset, error) {
+	resp, err := r.bc.Api.GetTableRows(eos.GetTableRowsRequest{
+		Code:  string(tokenContract),
+		Scope: string(playerAccount),
+		Table: "accounts",
+		JSON:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var jsonBalances []struct {
+		Balance eos.Asset `json:"balance"`
+	}
+	if err := resp.JSONToStructs(&jsonBalances); err != nil {
+		return nil, err
+	}
+	balances := make([]eos.Asset, len(jsonBalances))
+	for i, b := range jsonBalances {
+		balances[i] = b.Balance
+	}
+	return balances, nil
+}
+
+type TokenContract struct {
+	TokenName string          `json:"token_name"`
+	Contract  eos.AccountName `json:"contract"`
+}
+
+func (r *CasinoBlockchainRepo) getTokenToContract() (map[string]eos.AccountName, error) {
+	// TODO cache the response
+	resp, err := r.bc.Api.GetTableRows(eos.GetTableRowsRequest{
+		Code:  r.platformContract,
+		Scope: r.platformContract,
+		Table: "token",
+		JSON:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var jsonTokenContract []TokenContract
+	if err := resp.JSONToStructs(&jsonTokenContract); err != nil {
+		return nil, err
+	}
+	tokenContracts := make(map[string]eos.AccountName)
+	for _, tc := range jsonTokenContract {
+		tokenContracts[tc.TokenName] = tc.Contract
+	}
+	return tokenContracts, nil
+}
+
+func (r *CasinoBlockchainRepo) GetCustomTokenBalances(casinoName string,
+	accountName string) (map[string]eos.Asset, error) {
+	resp, err := r.bc.Api.GetTableRows(eos.GetTableRowsRequest{
+		Code:  casinoName,
+		Scope: casinoName,
+		Table: "token",
+		Limit: 1000,
+		JSON:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var jsonCasinoSupportedTokens []struct {
+		Name   string `json:"token_name"`
+		Paused int    `json:"paused"`
+	}
+	err = resp.JSONToStructs(&jsonCasinoSupportedTokens)
+	if err != nil {
+		return nil, err
+	}
+	casinoSupportedTokens := make(map[string]bool)
+	for _, token := range jsonCasinoSupportedTokens {
+		if token.Name != contracts.CoreSymbol {
```

Comment:
why can't we use BET as other custom tokens?

Reply:
we could but that would break the interface